garbage_classifier/garbage_classifier/garbage_data.py

```diff
@@ -34,22 +38,112 @@ def __len__(self):
+    _create_train_test_split(f'{folder}/input', f'{folder}/train', f'{folder}/test')
+    _create_tar_gz_folder(f'{folder}/train', 'data/real-ds/train.tar.gz')
+    _create_tar_gz_folder(f'{folder}/test', 'data/real-ds/test.tar.gz')
+
+def load_data(s3_access_key, s3_secret_key, s3_bucket, s3_prefix):
+    s3_client = boto3.client('s3', aws_access_key_id=s3_access_key, aws_secret_access_key=s3_secret_key)
+    folder = 'data/tmp'
+    _download_files_from_s3_bucket_folder(s3_client, s3_bucket, s3_prefix, f'{folder}')
+    _create_tar_gz_folder(f'{folder}', 'data/data.tar.gz')
+    
+def _create_train_test_split(src_folder, train_folder, test_folder):
+    class_folders = [f for f in os.listdir(src_folder) if os.path.isdir(os.path.join(src_folder, f))]
+
+    for class_folder in class_folders:
+        src_class_folder = os.path.join(src_folder, class_folder)
+        train_class_folder = os.path.join(train_folder, class_folder)
+        test_class_folder = os.path.join(test_folder, class_folder)
+
+        os.makedirs(train_class_folder, exist_ok=True)
+        os.makedirs(test_class_folder, exist_ok=True)
+
+        images = [f for f in os.listdir(src_class_folder) if os.path.isfile(os.path.join(src_class_folder, f))]
+        np.random.shuffle(images)
+        split_index = int(len(images) * 0.8)
+
+        train_images = images[:split_index]
+        test_images = images[split_index:]
+
+        for img in train_images:
+            shutil.copy(os.path.join(src_class_folder, img), os.path.join(train_class_folder, img))
+
+        for img in test_images:
+            shutil.copy(os.path.join(src_class_folder, img), os.path.join(test_class_folder, img))
+
+def _create_tar_gz_folder(src_folder, output_filename):
+    with tarfile.open(output_filename, "w:gz") as tar:
+        for entry in os.scandir(src_folder):
+            if entry.is_dir():
+                arcname = os.path.basename(entry.path)
+                tar.add(entry.path, arcname=arcname)
+
+
+def _download_files_from_s3_bucket_folder(s3, bucket_name, folder_path, local_destination):
+    paginator = s3.get_paginator('list_objects_v2')
+    operation_parameters = {'Bucket': bucket_name, 'Prefix': folder_path}
+
+    for page in paginator.paginate(**operation_parameters):
+        for item in page.get('Contents', []):
+            if not item['Key'].endswith('/'):
+                file_key = item['Key']
+                local_file_path = os.path.join(local_destination, os.path.relpath(file_key, folder_path))
+
+                os.makedirs(os.path.dirname(local_file_path), exist_ok=True)
+                print(f'Downloading {file_key} to {local_file_path}')
+                s3.download_file(bucket_name, file_key, local_file_path)
+                print(f'Downloaded {file_key} to {local_file_path}')
+
+def extract_tar_gz(archive_path):
+    if not os.path.exists(archive_path):
+        print(f"The file '{archive_path}' does not exist.")
+        return None
```

Comment:
don't need to return None, python does this by default

Reply:
I think GPT did it to terminate execution :)

Reply:
Ah, I see now